Core/HTTPSUpgradeParser.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  HTTPSUpgradeParser.swift
+//  DuckDuckGo
+//
+//  Copyright © 2018 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public class HTTPSUpgradeParser {
+
+    static func whitelist(fromJSONData data: Data) -> [String]? {
+        guard let json = try? JSONSerialization.jsonObject(with: data, options: []) else { return nil }
+        guard let jsonObject = json as? [String: Any] else { return nil }
+        guard let domains = jsonObject["data"] as? [String] else { return nil }
+        return domains
+    }
+    
+    static func bloomFilterSpecification(fromJSONData data: Data) -> HTTPSTransientBloomFilterSpecification? {
+        guard let json = try? JSONSerialization.jsonObject(with: data, options: []) else { return nil }
```

Comment:
If `HTTPSTransientBloomFilterSpecification` was `Decodable` you could just use the `JSONDecoder`.  Likewise for this whitelist.

We've got a couple of examples in the code already, though we also have examples of `JSONSeralization`, but I'd like us to move to the `Codable` pattern as our standard.  I'll add a LHF task to update them.

Reply:
Good suggestion. I had considered using Codable but was refactoring existing code and didn't know we had it in the project. I feel better using the modern pattern here, thanks!